Add script to parse through data for external links to search through…

kurtw opened the pull request:
We need icons for links to external sites. This adds the markup for the icons to sheer indexed content, as well as refactors the script to do the processing.

## Changes

- BeautifulSoup 4.4.1
- The script has been refactored to use more robust bs4 methods
- All sheer processors use the external links script now

## Testing

- Run `pip install -r requirements/local.txt`.
- Run `./cfgov/manage.py sheer_index`.
- Run `gulp build`
- Check out [this page](http://localhost:8000/blog/5-things-to-consider-before-you-collect-your-social-security-benefits/) for the icon on the `seven in ten consumers` link but not others.
- Be sure to check out the Wagtail richtextfield. I couldn't easily find a richtextfield in wagtail with an external link but you could go ahead and add that yourself.  

## Review

- @kave 
- @richaagarwal 
- @rosskarchner 
- @jimmynotjim 

## Screenshots
![screen shot 2016-03-22 at 11 09 29 am](https://cloud.githubusercontent.com/assets/1412978/13966334/3c09ebae-f048-11e5-92f6-759dabe451ec.png)

sebworks commented on cfgov/v1/__init__.py:
```diff
@@ -39,38 +40,29 @@ def environment(**options):
+    a_class = os.environ.get('EXTERNAL_A_CSS', 'icon-link icon-link__external-link')
+    span_class = os.environ.get('EXTERNAL_SPAN_CSS', 'icon-link_text')
+    for a in soup.find_all('a', href=True):
+        if p.match(a['href']):
```
Can we update the href as well to point to the external redirect screen? This code, combined with the Wagtail redirect code, would allow use to delete the JS that is handling the redirects. It would also still work if JS is disabled?

kurtw replied:
YES that would be awesome.

richaagarwal commented:
:+1:

kurtw commented:
Removed the external-site-redirect.js and replaced it with prepending it external links with the external site URL pattern. Ready for review again. Note: Added step to include running a `gulp build`.

sebworks commented:
:+1:

anselmbradford commented on cfgov/cfgov/settings/base.py:
```diff
@@ -186,6 +186,8 @@
 
 ALLOWED_HOSTS = ['*']
 
+EXTERNAL_LINK_PATTERN = r'https?:\/\/(?:www\.)?(?![^\?]+gov)(?!(content\.)?localhost).*'
```
This has been thoroughly tested, ya? I can't get it to pick up our domains in http://pythex.org/

anselmbradford replied:
Ooo it's not our domains, darf.. okay carryon.

anselmbradford commented on cfgov/jinja2/v1/_includes/macros/footer-global.html:
```diff
@@ -119,7 +120,7 @@
                                   icon-link
                                   icon-link__external-link"
-                           href="http://www.federalreserve.gov/oig/default.htm">
+                           href="{{ external_redirect_url }}http://www.federalreserve.gov/oig/default.htm">
```
Actual URL doesn't need URL-escaping?

sebworks replied:
They currently aren't escaped but we can add it.
